### PR TITLE
[enc2]fixing recovery tooltip

### LINKED ISSUE #15822: [enc2] No error message for wrong recovery password in users page
### Steps
1. Enable encryption + default module
2. As "root" (the admin) add a recovery password "recovery" in the admin page
3. Create a user "user1" with the password "test"
4. Login as "user1" 
5. Upload a file
6. Login as "root"
7. Go to the users page
8. Enter the wrong recovery password "wrong"
9. Change "user1"'s password

### Expected result
Error message that recovery password is wrong (or that password could not be changed)

### Actual result
No UI feedback. Ajax call returns 200 OK.

### Versions
ownCloud git master (f8f354b351a349898bbb5cdf2d9bee1c798c0f73)

@schiesbn @th3fallen @DeepDiver1975

### PR DESCRIPTION
**Pull request by th3fallen**
resolves https://github.com/owncloud/core/issues/15822 and https://github.com/owncloud/core/issues/15823 and https://github.com/owncloud/core/issues/16196

### CODE REVIEW

**Comment by th3fallen** [comment]
@PVince81 @DeepDiver1975 @schiesbn

File: settings/controller/userscontroller.php [hunk 1]
```diff
@@ -122,7 +122,7 @@ public function __construct($appName,
 		$this->subAdminFactory = $subAdminFactory;
 
 		// check for encryption state - TODO see formatUserForIndex
-		$this->isEncryptionAppEnabled = $appManager->isEnabledForUser('files_encryption');
+		$this->isEncryptionAppEnabled = $appManager->isEnabledForUser('encryption');
```
Comment (PVince81) [inline settings/controller/userscontroller.php]: Did you grep the whole code for "files_encryption", in case there are other remains ?

Comment (th3fallen) [inline settings/controller/userscontroller.php]: yes i did but looks like i forgot to commit one somehow..

**Comment by PVince81** [comment]
Can you add a unit test ? This will certainly help to avoid this breaking again :smile:

**Comment by th3fallen** [comment]
im not sure that method is testable since all the encryption things are called within the method and no injected @PVince81 ideas if there is a way to get around that?

**Comment by PVince81** [comment]
It doesn't seem to work properly: I did enable recovery for "user1" but I see the red highlight and warning that the user didn't and data loss may happen.

Also entering the wrong recovery password doesn't show any additional error message.

**Comment by th3fallen** [comment]
is recovery enabled for the user your trying to get the wrong recovery password error on @PVince81

**Comment by PVince81** [comment]
I have two users: user1 has recovery enabled, user2 hasn't.
Expected result: warning for user2, not user1
Actual result: getting the warning for both user1 and user2

**Comment by PVince81** [comment]
As for the unit tests: I see that the config is injected, so you might be able to mock the "recoveryAdminEnabled" value and also the user's config settings that tells whether recovery is enabled.

The matching unit test class seems to already have a test for "index()", which uses the format calls. You could extend that one and have one of the users have recovery enabled, then check if the data response has "isRestoreDisabled" => false for one and true for the other one.

**Comment by th3fallen** [comment]
> I have two users: user1 has recovery enabled, user2 hasn't.
> Expected result: warning for user2, not user1
> Actual result: getting the warning for both user1 and user2

if that's the case thats how it worked in the prior version i simply changed the app name so it could recognize it again. I didn't change any logic.

**Comment by PVince81** [comment]
We still miss a unit test that sets that config value and verifies that the rest works well (and that value must be tested with both "false" and "true")
If modifying code doesn't break any existing unit test, it means there was no proper unit test in the first place.

**Comment by PVince81** [comment]
Ah sorry, I thought you were talking about the unit test.
Let me verify again how it worked on stable8  then.

**Comment by PVince81** [comment]
@th3fallen I tried stable8 (b453cccf51598a45d327ef40ee5298a01b233dbf) and it works as I described: user1 doesn't get a warning, user2 does.

**Comment by PVince81** [comment]
and entering the wrong recovery password, then changing user1's password shows a notification "Wrong admin recovery password". It doesn't happen on this PR either.

Maybe some values are not updated/passed properly to the client ?

**Comment by th3fallen** [comment]
you're right found it, and corrected.

**Comment by th3fallen** [comment]
@owncloud-bot retest this please

**Comment by DeepDiver1975** [comment]
squashed ....

**Comment by DeepDiver1975** [comment]
@owncloud-bot retest this please

**Comment by th3fallen** [comment]
great now you broke him @DeepDiver1975 :wink:

**Comment by schiesbn** [comment]
I just tried it. https://github.com/owncloud/core/issues/15823 works great. But I still don't get a error if the recovery key password was wrong

File: settings/changepassword/controller.php [hunk 2]
```diff
@@ -115,28 +115,25 @@ public static function changeUserPassword($args) {
 				$validRecoveryPassword = $keyManager->checkRecoveryPassword($recoveryPassword);
 				$recoveryEnabledForUser = $recovery->isRecoveryEnabledForUser();
```
Comment (schiesbn) [inline settings/changepassword/controller.php]: need to add the user for whom we want to change the password as argument.

Comment (th3fallen) [inline settings/changepassword/controller.php]: great catch! should be fixed.

**Comment by schiesbn** [comment]
Tested, now it works as expected!

But on my local test run three tests failed:

```
There were 3 failures:

1) OC\Settings\Controller\UsersControllerTest::testRestorePossibleWithAdminAndUserRestore
Expectation failed for method name is equal to <string:isEnabledForUser> when invoked 1 time(s)
Parameter 0 for invocation OCP\App\IAppManager::isEnabledForUser('encryption', null) does not match expected value.
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'files_encryption'
+'encryption'

/home/src/owncloud/server/master/settings/controller/userscontroller.php:125
/home/src/owncloud/server/master/settings/application.php:136
/home/src/owncloud/server/master/3rdparty/pimple/pimple/src/Pimple/Container.php:112
/home/src/owncloud/server/master/tests/settings/controller/userscontrollertest.php:1330

2) OC\Settings\Controller\UsersControllerTest::testRestoreNotPossibleWithoutAdminRestore
Expectation failed for method name is equal to <string:isEnabledForUser> when invoked zero or more times
Parameter 0 for invocation OCP\App\IAppManager::isEnabledForUser('encryption', null) does not match expected value.
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'files_encryption'
+'encryption'

/home/src/owncloud/server/master/settings/controller/userscontroller.php:125
/home/src/owncloud/server/master/settings/application.php:136
/home/src/owncloud/server/master/3rdparty/pimple/pimple/src/Pimple/Container.php:112
/home/src/owncloud/server/master/tests/settings/controller/userscontrollertest.php:1348

3) OC\Settings\Controller\UsersControllerTest::testRestoreNotPossibleWithoutUserRestore
Expectation failed for method name is equal to <string:isEnabledForUser> when invoked 1 time(s)
Parameter 0 for invocation OCP\App\IAppManager::isEnabledForUser('encryption', null) does not match expected value.
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'files_encryption'
+'encryption'

/home/src/owncloud/server/master/settings/controller/userscontroller.php:125
/home/src/owncloud/server/master/settings/application.php:136
/home/src/owncloud/server/master/3rdparty/pimple/pimple/src/Pimple/Container.php:112
/home/src/owncloud/server/master/tests/settings/controller/userscontrollertest.php:1387
````

**Comment by schiesbn** [comment]
@th3fallen any progress here? We can close quite some tickets if this pr is merged?

**Comment by th3fallen** [comment]
all fixed.

**Comment by schiesbn** [comment]
seems like we need a new pull request so that Jenkins can run the tests successfully.

**Comment by schiesbn** [comment]
Created a extra PR for Jenkins... Let's see if all tests pass https://github.com/owncloud/core/pull/16274

**Comment by schiesbn** [comment]
all tests passed... :+1: from my side

**Comment by schiesbn** [comment]
@PVince81 @SergioBertolinSG this fixes some issues created by both of you. Can you verify that it fixes the issues? Thanks!

**Comment by PVince81** [comment]
@owncloud-bot retest this please

**Comment by PVince81** [comment]
Works now :+1:

**Comment by PVince81** [comment]
Jenkins PR: https://github.com/owncloud/core/pull/16377

**Pull request merged**